servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/Http2Exception.java

```diff
@@ -35,14 +39,36 @@
 
     static Throwable wrapIfNecessary(final Throwable cause) {
         if (cause instanceof io.netty.handler.codec.http2.Http2Exception) {
-            return new StacklessHttp2Exception((io.netty.handler.codec.http2.Http2Exception) cause);
+            final io.netty.handler.codec.http2.Http2Exception h2Cause =
+                    (io.netty.handler.codec.http2.Http2Exception) cause;
+            return isRetryable(h2Cause) ? new RetryableStacklessHttp2Exception(h2Cause) :
+                    new StacklessHttp2Exception(h2Cause);
         }
         if (cause instanceof io.netty.handler.codec.http2.Http2FrameStreamException) {
             return new StacklessHttp2Exception((io.netty.handler.codec.http2.Http2FrameStreamException) cause);
         }
         return cause;
     }
 
+    /**
+     * Checks if an {@link io.netty.handler.codec.http2.Http2Exception} is retryable on a different h2 parent
+     * connection.
+     *
+     * @param cause {@link io.netty.handler.codec.http2.Http2Exception} for inspection
+     * @return {@code true} if {@link io.netty.handler.codec.http2.Http2Exception} is retryable on a different h2
+     * parent connection.
+     */
+    private static boolean isRetryable(final io.netty.handler.codec.http2.Http2Exception cause) {
+        // The first check captures cases like:
+        //  - Cannot create stream %d greater than Last-Stream-ID %d from GOAWAY.
+        //  - Stream IDs are exhausted for this endpoint.
+        //  - Maximum active streams violated for this endpoint.
+        //  - Http2ChannelClosedException
+        return cause.error() == REFUSED_STREAM
```

Comment:
There is code in `AbstractH2DuplexHandler` that also checks for `REFUSED_STREAM` and throws a `Retryable` exception. is there a way to have these two areas share code so the relationship is more clear and less likely to miss cases in the future (e.g. if we want to add `ENHANCE_YOUR_CALM` or others)?

Reply:
I thought about it but did not found a good way to do that because in `AbstractH2DuplexHandler` it handles a user-event and created a new exception based on that event when here we intercept write failures. May give another round of thoughts after the release.

Reply:
Follow-up with the fresh head :)
https://github.com/apple/servicetalk/pull/1318